### PR TITLE
Fix intermittent test failure.

### PR DESCRIPTION
**Pull request by howbazaar**
The TestClientWatchAllAdminPermission from apiserver/facades/client/client failed intermittently.

When run locally it would fail about 50% of the time. After much digging, I found that this was a race in the underlying multiwatcher code. If the intial GetAll was processed and the underlying changes from the document watchers hadn't been processed, the test would pass, but if the change for the remote application was processed before Next was called in the test, it would fail.

This was due to the entity being assigned into the multiwatcher state was missing the status value. To handle this, if the entity isn't new, we get the status from the old value.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1790548

### CODE REVIEW

**Comment by howbazaar** [comment]
$$merge$$

**Comment by howbazaar** [comment]
$$merge$$

File: state/allwatcher_internal_test.go [hunk 1]
```diff
@@ -1028,11 +990,12 @@ func (s *allWatcherStateSuite) TestStateWatcher(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	err = wu.AssignToMachine(m2)
 	c.Assert(err, jc.ErrorIsNil)
+	wpTime := s.state.clock().Now()
 
 	// Look for the state changes from the allwatcher.
 	deltas = tw.All(5)
 
-	zeroOutTimestampsForDeltas(c, deltas)
+	// zeroOutTimestampsForDeltas(c, deltas)
```
Comment (wallyworld) [inline state/allwatcher_internal_test.go]: delete me

File: state/allwatcher_internal_test.go [hunk 2]
```diff
@@ -1834,7 +1792,7 @@ func (s *allModelWatcherStateSuite) TestStateWatcher(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 
 	deltas = tw.All(1)
-	zeroOutTimestampsForDeltas(c, deltas)
+	//zeroOutTimestampsForDeltas(c, deltas)
```
Comment (wallyworld) [inline state/allwatcher_internal_test.go]: delete me

File: state/allwatcher_internal_test.go [hunk 3]
```diff
@@ -1860,7 +1818,7 @@ func (s *allModelWatcherStateSuite) TestStateWatcher(c *gc.C) {
...
```
Comment (wallyworld) [inline state/allwatcher_internal_test.go]: etc etc

**Comment by howbazaar** [comment]
4 other intermittent failures.
$$merge$$

**Pull request merged**